src/order/heyting/hom.lean

```diff
@@ -0,0 +1,405 @@
+  [coheyting_algebra β] [coheyting_hom_class F α β] : bounded_lattice_hom_class F α β :=
+{ map_bot := λ f, by rw [←@sdiff_self α _ ⊤, ←sdiff_self, map_sdiff],
+  ..‹coheyting_hom_class F α β› }
+
+@[priority 100] -- See note [lower instance priority]
+instance biheyting_hom_class.to_heyting_hom_class [biheyting_algebra α] [biheyting_algebra β]
+  [biheyting_hom_class F α β] :
+  heyting_hom_class F α β :=
+{ map_bot := λ f, by rw [←@sdiff_self α _ ⊤, ←sdiff_self, biheyting_hom_class.map_sdiff],
+  ..‹biheyting_hom_class F α β› }
+
+@[priority 100] -- See note [lower instance priority]
+instance biheyting_hom_class.to_coheyting_hom_class [biheyting_algebra α] [biheyting_algebra β]
+  [biheyting_hom_class F α β] :
+  coheyting_hom_class F α β :=
+{ map_top := λ f, by rw [←@himp_self α _ ⊥, ←himp_self, map_himp],
+  ..‹biheyting_hom_class F α β› }
+
+@[priority 100] -- See note [lower instance priority]
+instance order_iso_class.to_heyting_hom_class [heyting_algebra α] [heyting_algebra β]
+  [order_iso_class F α β] :
+  heyting_hom_class F α β :=
+{ map_himp := λ f a b, eq_of_forall_le_iff $ λ c,
+    by { simp only [←map_inv_le_iff, le_himp_iff], rw ←order_iso_class.map_le_map_iff f, simp },
+  ..order_iso_class.to_bounded_lattice_hom_class }
+
+@[priority 100] -- See note [lower instance priority]
+instance order_iso_class.to_coheyting_hom_class [coheyting_algebra α] [coheyting_algebra β]
+  [order_iso_class F α β] :
+  coheyting_hom_class F α β :=
+{ map_sdiff := λ f a b, eq_of_forall_ge_iff $ λ c,
+    by { simp only [←le_map_inv_iff, sdiff_le_iff], rw ←order_iso_class.map_le_map_iff f, simp },
+  ..order_iso_class.to_bounded_lattice_hom_class }
+
+@[priority 100] -- See note [lower instance priority]
+instance order_iso_class.to_biheyting_hom_class [biheyting_algebra α] [biheyting_algebra β]
+  [order_iso_class F α β] :
+  biheyting_hom_class F α β :=
+{ map_himp := λ f a b, eq_of_forall_le_iff $ λ c,
+    by { simp only [←map_inv_le_iff, le_himp_iff], rw ←order_iso_class.map_le_map_iff f, simp },
+  map_sdiff := λ f a b, eq_of_forall_ge_iff $ λ c,
+    by { simp only [←le_map_inv_iff, sdiff_le_iff], rw ←order_iso_class.map_le_map_iff f, simp },
+  ..order_iso_class.to_lattice_hom_class }
+
+/-- This can't be an instance because of typeclass loops. -/
+@[reducible] -- See note [reducible non instances]
+def bounded_lattice_hom_class.to_biheyting_hom_class [boolean_algebra α] [boolean_algebra β]
+  [bounded_lattice_hom_class F α β] :
+  biheyting_hom_class F α β :=
+{ map_himp := λ f a b, by rw [himp_eq, himp_eq, map_sup, (is_compl_compl.map _).compl_eq],
+  map_sdiff := λ f a b, by rw [sdiff_eq, sdiff_eq, map_inf, (is_compl_compl.map _).compl_eq],
+   ..‹bounded_lattice_hom_class F α β› }
+
+section heyting_algebra
+variables [heyting_algebra α] [heyting_algebra β] [heyting_hom_class F α β] (f : F)
+include β
+
+@[simp] lemma map_compl (a : α) : f aᶜ = (f a)ᶜ := by rw [←himp_bot, ←himp_bot, map_himp, map_bot]
+
+-- TODO: `map_bihimp`
```

Comment:
What is the status of this TODO? Waiting on follow-up PRs?

Reply:
Yes, the introduction of `bihimp` is waiting on #16282.